dashboard/app/controllers/script_levels_controller.rb

```diff
@@ -22,12 +22,16 @@ def reset
     # delete the client state and other session state if the user is not signed in
     # and start them at the beginning of the script.
     # If the user is signed in, continue normally.
-    unless current_user
+    redirect_url = build_script_level_path(@script.starting_level)
```

Comment:
`build_script_level_path` returns a relative path not a url (there is `build_script_level_url` if you want a full url) so maybe rename the variable to redirect_path to make that clear (in this case, where you use it either a path or a url will work so it's not a big deal)